lib/torrent.js

```diff
@@ -600,12 +606,18 @@ class Torrent extends EventEmitter {
             this._debug('piece verified %s', index)
             this._markVerified(index)
           } else {
+            this.pieces[index] = new Piece(length || this.pieceLength)
+            this.bitfield.set(index, false)
+            this.select(index, index, 1)
+            this.files.forEach(file => {
+              if (file.done && file.includes(index)) file.done = false
+            })
             this._debug('piece invalid %s', index)
```

Comment:
@jhiesey this part is useful to mark the current torrent that the piece found in the store is not correct and should be re-requested.    Helps keep the torrent consistent

Reply:
@alxhotel  too :)